kahuna/public/js/services/api/collections-api.js

```diff
@@ -64,6 +66,41 @@ apiServices.factory('collections', ['mediaApi', function (mediaApi) {
         return Promise.all(promises);
     }
 
+    function collectionsEquals(collectionsA, collectionsB) {
+        return angular.equals(
+            collectionsA.sort(),
+            collectionsB.sort()
+        );
+    }
+
+    function getCollectionsIds(imageCollections) {
+        const collectionsIds = [];
+        imageCollections.data.forEach(e => collectionsIds.push(e.pathId));
+        return collectionsIds;
+    }
+
+    function untilCollectionsEqual(image, expectedCollections) {
+        return image.get().then(apiImage => {
```

Comment:
I know this seems to be the de facto way of doing this - but collections don't really need to poll as they don't affect image state - so I wonder if there is another way to update the collections.

Saying that, maybe it's worth leaving and doing in one refactor when we get to it @theefer?